### PR TITLE
[REVIEW] Use documented header template for `doxygen`

### PR DESCRIPTION
**Pull request by galipremsagar**
This PR uses a documented template auto-generated by `doxygen` and inserts our custom js & css links to it. The process is documented here: https://doxygen.nl/manual/customize.html#minor_tweaks_header_css

xref: https://github.com/rapidsai/cudf/pull/11430

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]

File: cpp/doxygen/header.html
Comment (isVoid) [inline cpp/doxygen/header.html:1]: Sorry I'm a bit late to the review process. The content of this file doesn't seem specific to cuspatial (and other RAPIDS library). Any reason this cannot be included in rapidsai/docs?

I thought about auto generating this file in the sphinx's build process and automate the injection of the RAPIDS actions and stylesheets. But perhaps a bit brittle if doxygen updates their generate process.

Comment (galipremsagar) [reply cpp/doxygen/header.html:1]: > Sorry I'm a bit late to the review process. The content of this file doesn't seem specific to cuspatial (and other RAPIDS library). Any reason this cannot be included in rapidsai/docs?

That's unfortunately not possible because the doxygen builds happen at the repository level and not in `rapidsai/docs`.

**Review by harrism — APPROVED** [review]
Thanks for adding this to cuSpatial!

**Comment by harrism** [comment]
@galipremsagar is this ready to merge?

**Comment by galipremsagar** [comment]
> @galipremsagar is this ready to merge?

Yes Mark, was waiting for an approval.

**Comment by galipremsagar** [comment]
@gpucibot merge

**Comment by galipremsagar** [comment]
> @gpucibot merge

Don't think I have permissions, could you merge it @harrism ?

**Comment by harrism** [comment]
@gpucibot merge

**Comment by galipremsagar** [comment]
Thanks Mark!

**Pull request merged**
